frame/identity/src/lib.rs

```diff
@@ -714,8 +907,13 @@ decl_module! {
 		/// # <weight>
 		/// - `O(R)`.
 		/// - One storage mutation `O(R)`.
+		/// - Benchmarks:
+		///   - 23.81 + R * 0.774 µs (min squares analysis)
 		/// # </weight>
-		#[weight = 50_000_000]
+		#[weight = T::DbWeight::get().reads_writes(1, 1)
+			+ 24_000_000 // constant
+			+ 780_000 * T::MaxRegistrars::get() as Weight // R
+		]
 		fn set_fee(origin,
```

Comment:
Why no refund on this?

Reply:
I thought it would complicate the code a lot, will push a commit implementing it and we can see.

Reply:
there were some others that could use refunds too, basically anything you didnt put in the function block.

Would be good to know if it does significantly complicate the logic (for example if it invokes a new storage read for whatever reason)

Reply:
No, it's just more code.